### PR TITLE
[3.4.2]Feature: New ruleNode 'json path'

### PR DESCRIPTION
**Pull request by YuriyLytvynchuk**
## Pull Request description
issue: https://github.com/thingsboard/thingsboard/issues/7100
UI generator PR: https://github.com/thingsboard/thingsboard-rule-config-ui-ngx/pull/88

## General checklist

- [x] You have reviewed the guidelines [document](https://docs.google.com/document/d/1wqcOafLx5hth8SAg4dqV_LV3un3m5WYR8RdTJ4MbbUM/edit?usp=sharing).
- [x] PR name contains fix version. For example, "[3.3.4] Hotfix of some UI component" or "[3.4] New Super Feature".
- [x] Description references specific [issue](https://github.com/thingsboard/thingsboard/issues).
- [x] Description contains human-readable scope of changes.
- [x] Description contains brief notes about what needs to be added to the documentation.
- [x] No merge conflicts, commented blocks of code, code formatting issues.
- [x] Changes are backward compatible or upgrade script is provided.
- [x] Similar PR is opened for PE version to simplify merge. Required for internal contributors only.
  
## Front-End feature checklist

- [x] Screenshots with affected component(s) are added. The best option is to provide 2 screens: before and after changes;
- [x] If you change the widget or other API, ensure it is backward-compatible or upgrade script is present.
- [x] Ensure new API is documented [here](https://github.com/thingsboard/thingsboard-ui-help)

## Back-End feature checklist

- [x] Added corresponding unit and/or integration test(s). Provide written explanation in the PR description if you have failed to add tests.
- [x] If new dependency was added: the dependency tree is checked for conflicts.
- [x] If new service was added: the service is marked with corresponding @TbCoreComponent, @TbRuleEngineComponent, @TbTransportComponent, etc.
- [x] If new REST API was added: the RestClient.java was updated, issue for [Python REST client](https://github.com/thingsboard/thingsboard-python-rest-client) is created.

### CODE REVIEW

**Review by ShvaykaD — APPROVED** [review]

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    TbJsonPathNodeConfiguration config;
+    Configuration configurationJsonPath;
+    JsonPath jsonPath;
+    String jsonPathValue;
+
+    @Override
+    public void init(TbContext ctx, TbNodeConfiguration configuration) throws TbNodeException {
+        this.config = TbNodeUtils.convert(configuration, TbJsonPathNodeConfiguration.class);
+        this.jsonPathValue = config.getJsonPath();
+        if (this.jsonPathValue != "$") {
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java:60]: I think that it's better to add equals here and make it reverse to handle null values

Comment (ShvaykaD) [reply rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java:60]: "$" can be added to the constant in the configuration class

File: rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/transform/TbJsonPathNodeTest.java [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+    void givenDefaultConfig_whenInit_thenFail() {
+        config.setJsonPath("");
+        nodeConfiguration = new TbNodeConfiguration(mapper.valueToTree(config));
+        assertThatThrownBy(() -> node.init(ctx, nodeConfiguration)).isInstanceOf(IllegalArgumentException.class);
+    }
+
+    @Test
+    void givenDefaultConfig_whenVerify_thenOK() {
+        TbJsonPathNodeConfiguration defaultConfig = new TbJsonPathNodeConfiguration().defaultConfiguration();
+        assertThat(defaultConfig.getJsonPath()).isEqualTo("$");
```
Comment (ShvaykaD) [inline rule-engine/rule-engine-components/src/test/java/org/thingsboard/rule/engine/transform/TbJsonPathNodeTest.java:89]: change "$" to constant from config class

**Review by ShvaykaD — APPROVED** [review]

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.TRANSFORMATION,
+        name = "json path",
+        configClazz = TbJsonPathNodeConfiguration.class,
+        nodeDescription = "Extracts json element or set of elements from a message by JSONPath expression.",
+        nodeDetails = "JSONPath expression specifies a path to an element or a set of elements in a JSON structure. <br/>"
+                + "<b>'$'</b> represents the root object or array. If JSONPath expression evaluation failed, incoming "
```
Comment (ashvayka) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java:43]: Separate last sentence with <br/><br/>

File: rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+import org.thingsboard.server.common.msg.TbMsg;
+
+import java.util.concurrent.ExecutionException;
+
+@Slf4j
+@RuleNode(
+        type = ComponentType.TRANSFORMATION,
+        name = "json path",
+        configClazz = TbJsonPathNodeConfiguration.class,
+        nodeDescription = "Extracts json element or set of elements from a message by JSONPath expression.",
```
Comment (ashvayka) [inline rule-engine/rule-engine-components/src/main/java/org/thingsboard/rule/engine/transform/TbJsonPathNode.java:41]: Transforms incoming message body using JSONPath expression

**Review by ShvaykaD — APPROVED** [review]

**Pull request merged**